src/test/java/FlightsReporterTests.groovy

```diff
@@ -0,0 +1,197 @@
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.SOFIA), 460)
+        assertEquals(FlightsReporter.passengerAmountFrom(flights, FlightsReporter.Airports.NEW_YORK), 0)
+    }
+
+    @Test
+    void testSortingByDepartureDate() {
+        def departureExpected = []
+        def fl1 = flights.collect()
+        departureExpected << flights[9] << flights[5] << flights[1] << flights[0] << flights[4] << flights[6]
+                << flights[2] << flights[3] << flights[8] << flights[7]
+        FlightsReporter.sortByDepartureDate(fl1)
+        assertEquals(fl1, departureExpected)
+    }
+
+    @Test
+    void testSortingByArrivalDate() {
+        def arrivalExpected = []
+        def fl2 = flights.collect()
+        arrivalExpected << flights[9] << flights[1] << flights[0] << flights[4] << flights[5] << flights[6]
+                << flights[2] << flights[3] << flights[8] << flights[7]
+        FlightsReporter.sortByArrivalDate(fl2)
+        assertEquals(fl2, arrivalExpected)
+    }
+
+    @Test
+    void testSortingByFlightLen() {
+        def flightLenExpected = []
+        def fl1 = flights.collect()
+        flightLenExpected << flights[0] << flights[7] << flights[4] << flights[1] << flights[2] << flights[9]
+                << flights[3] << flights[6] << flights[8] << flights[5]
+        FlightsReporter.sortByFlightLength(fl1)
+        assertEquals(fl1, flightLenExpected)
+    }
+
+    @Test
+    void testGetAllPlanesAboveCapacity() {
+        def expected = []
+        expected << flights[0] << flights[5] << flights[6]
+        assertEquals(FlightsReporter.getAllPlanesAboveCertainCapacityFull(flights, 0.99), expected)
+    }
+
+    @Test
+    void testRunningLate() {
+        assertEquals(true, FlightsReporter.isThereAFlightRunningLate(flights))
+
+        def flightsLate = []
+        flightsLate << flights[0] << flights[3] << flights[4] << flights[5] << flights[8] << flights[9]
+        assertEquals(flightsLate, FlightsReporter.getFlightsRunningLate(flights))
+    }
+
+    @Test
+    void testDepartingAfter() {
+        assertEquals(
+                FlightsReporter.getAllFlightsDepartingBefore(flights,
+                        LocalDateTime.of(2021, 7, 31, 10, 0)), [])
+
+        assertEquals(FlightsReporter.getAllFlightsDepartingBefore(flights,
+                        LocalDateTime.of(2021, 8, 2, 12, 0)),
+                        [flights[0], flights[1], flights[2], flights[4], flights[5], flights[6], flights[9]])
+    }
+
+
+
+    @Before
+    void initializeTestData() {
+        flights << new Flight(id: "1",
+                from: FlightsReporter.Airports.PARIS,
+                destination: FlightsReporter.Airports.AMSTERDAM,
+                departure: LocalDateTime.of(2021, 8, 1, 10, 0),
+                arrival: LocalDateTime.of(2021, 8, 1, 11, 10),
+                passengers: 150, capacity: 150, runningLate: true)
+
+                << new Flight(id: "2",
```

Comment:
If you are going to use all these 10 flights, please split them into different rows (statements). If one of these throws and exception, it will display row 127 and you wouldn't have details.

Lines have a max size of 80-120characters for a reason. Check IDEA for a vertical line